config/git_fetcher_test.go

```diff
@@ -0,0 +1,25 @@
+package config
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetCanonicalization(t *testing.T) {
+	vals := map[string][]string{
+		"user.name":                   []string{"Pat Doe"},
+		"branch.MixedCase.pushremote": []string{"Somewhere"},
```

Comment:
Here's a compelling case where we _wouldn't_ want the warning about uppercase keys... maybe we don't want it at all?

Reply:
Ah, but in your code, we never take the branch that prints the warning, because we have at least three components. I don't think this case makes a compelling argument for or against your proposed solution.